firebase-functions/ktx/src/main/kotlin/com/google/firebase/functions/ktx/Functions.kt

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.functions.ktx
+
+import android.support.annotation.Keep
+import com.google.firebase.FirebaseApp
+import com.google.firebase.functions.FirebaseFunctions
+import com.google.firebase.functions.HttpsCallableReference
+import com.google.firebase.components.Component
+import com.google.firebase.components.ComponentRegistrar
+
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.platforminfo.LibraryVersionComponent
+
+/** Returns the [FirebaseFunctions] instance of the default [FirebaseApp]. */
+val Firebase.functions: FirebaseFunctions
+    get() = FirebaseFunctions.getInstance()
+
+/** Returns the [FirebaseFunctions] instance of a given [region]. */
+fun Firebase.functions(region: String): FirebaseFunctions = FirebaseFunctions.getInstance(region)
+
+/** Returns the [FirebaseFunctions] instance of a given [FirebaseApp]. */
+fun Firebase.functions(app: FirebaseApp): FirebaseFunctions = FirebaseFunctions.getInstance(app)
+
+/** Returns the [FirebaseFunctions] instance of a given [FirebaseApp] and [region]. */
+fun Firebase.functions(app: FirebaseApp, region: String): FirebaseFunctions =
+        FirebaseFunctions.getInstance(app, region)
+
+/** Runs the callable. */
+operator fun HttpsCallableReference.invoke() = this.call()
```

Comment:
Hi @the-dagger @rosariopfernandes @SUPERCILEX I would like to get your feedback on this use of operator overload. It enables this use case

```
var myFunction = functions.getHttpsCallable("myFunction")
myFunction(data)
```

instead of 

```
var myFunction = functions.getHttpsCallable("myFunction")
myFunction.call(data)
```

Does it feel kotlin-idiomatic, or it may produce unwanted confusion?

Reply:
Hmmm, I think that might be overdoing it. Usually, the invoke overload is used to eliminate a method call that takes a lambda:
```
// Without
gradleTask.configure { ... }

// Overload
gradleTask { ... }
```

Would be curious to hear what other people think though.

Reply:
> Does it feel kotlin-idiomatic?

I'm not sure, because most times I find it being used with lambda functions, as @SUPERCILEX mentioned. But I like the fact that this makes it more similar to the JavaScript SDK.

> may produce unwanted confusion?

I don't think so, since `call` is the only action you can take from an instance of `HttpsCallableReference`.

Reply:
Thanks for the feedback @rosariopfernandes, @SUPERCILEX. I'll remove the overload as I tend to agree with you about using invoke mostly for cases that include lambdas as argument.